File: Data Structures & Algorithms/climbing-stairs/submission-1.js
class Solution {
    /**
     * @param {number} n
     * @return {number}
     */
    climbStairs(n) {
        let count = 0; 
        const dp = (idx) => {
            if(idx > n) return; 
            if(idx == n){
                count++;
                return
            }

            dp(idx + 1)
            dp(idx + 2)
        }

        dp(0)
        return count; 
    }
}

File: Data Structures & Algorithms/majority-element/submission-1.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        len_nums = len(nums)
        
        # Edge cases - nice to haves
        if len_nums < 1: 
            return 0
        if len_nums == 1:
            return nums[0]

        count, val = 1, nums[0]

        for idx in range(1, len_nums):
            curr = nums[idx]
            if val == curr:
                count += 1
            elif count == 1 and val != curr:
                val = curr
                count = 1
            else:
                count -= 1
        
        return val

        